### DISCUSSION TITLE
Configurable default sort order in a Table View & Pivot Table View

### DISCUSSION THREAD

**Discussion by polmonso** [apache/superset#30145] [Ideas]
I would like a way to sort Table View based on an arbitrary column and to be able to choose the order as either descending or ascending.

On 2016 @verysmallriver opened  #1523 asking for this but went stale.

On April 2016, @jacopoch and @mistercrunch wrote on gitter:

    Jacopo Chiapparino @jacopoch Apr 25 21:32
    Hi guys, I was wondering if there's any way to currently sort Viz Type "table" via the "Params" in the edit slice panel? I'm looking into pre-sorting a table in a specific way when the dashboard/slice is loaded (like, first column of the table, desc). Any help appreciated! Thanks!

    Maxime Beauchemin @mistercrunch Apr 27 13:43
    @jacopoch sorry about the delay, I think at the moment the default is to sort on the first metric in the table
    the row limit also applies based on that sort order
    if it's important to you, you can create a github issue

@maver1ck asked the same question on Sept 2016

    Maciej Bryński @maver1ck Sep 23 00:56
    Hi,
    I've got a question
    How can I specify sorting for table vizualization ?
    It defaults to descending on first aggregation
    Is there any option to change it ?

This still doesn't seem doable.

As you can see in the image below, the dates are all out of order.

![image](https://github.com/user-attachments/assets/539c8228-5e46-4fb3-9365-0ed014282270)

The Sort By field is the metric to decide the pagination order, not the display order. By pressing the ordering arrows we can achieve the desired result, but there's no way to persist this.

If instead of aggregate query mode we switch to `raw records`, there we do have an ordering option.

![image](https://github.com/user-attachments/assets/c6117e34-bd2b-4907-87b9-4fbe9be520be)

The feature request is to also have an ordering field on the aggregate query mode.

Superset 3.0.2

**Comment by swapnil0545** [comment]
I know this is pretty old, but is there any solution if we want to implement a default sort for any of the column in aggregate table?
